scripts/Convert-ProwlarrSupportedIndexersToMarkdownTable.ps1

```diff
@@ -0,0 +1,280 @@
+$markdown_escape_regex_rep = '$1\$2$3'
+$wiki_1newline = "`r`n"
+$wiki_2newline = "`r`n`r`n"
+$wiki_encoding = 'utf8'
+### Github Details
+$gh_web = 'https://github.com'
+$gh_web_commit = 'commit/'
+### Prowlarr Search String Dictionary
+$SearchTypes = @{
+    'Q'         = 'Text Query'
+    'Album'     = 'Album'
+    'Artist'    = 'Artist'
+    'Author'    = 'Author'
+    'Ep'        = 'Episode'
+    'Genre'     = 'Genre'
+    'ImdbId'    = 'IMDb ID'
+    'ImdbTitle' = 'IMDb Title'
+    'ImdbYear'  = 'IMDb Year'
+    'Label'     = 'Label'
+    'RId'       = 'TV Rage Id'
+    'Season'    = 'Season'
+    'Title'     = 'Title'
+    'TmdbId'    = 'TMDb Id'
+    'TraktId'   = 'Trakt Id'
+    'TvdbId'    = 'TVDb Id'
+    'TvMazeId'  = 'TV Maze Id'
+    'Year'      = 'Year'
+}
+
+function Invoke-SearchReplace
+{
+    param (
+        [Parameter(Mandatory)]
+        [PSCustomObject]
+        $SearchValue
+    )
+    if ($SearchValue)
+    {
+        foreach ($dict in $SearchTypes)
+        {
+            $SearchValue = $SearchValue -replace $dict.Name, $dict.Value
+        }
+        return $SearchValue
+    }
+    
+    else
+    {
+        return 'Not Supported'
+    }
+}
+
+function Invoke-ConvertToMarkDownTable
+{
+    param (
+        [Parameter(Mandatory)]
+        [PSCustomObject]
+        $object
+    )
+    return $($object | Format-MarkdownTableTableStyle Indexer, Description, Language, 'Supports Raw Search', 'Search Types', 'TV Search Types', 'Movie Search Types', 'Music Search Types', 'Book Search Types', -HideStandardOutput -ShowMarkdown -DoNotCopyToClipboard )
+}
+
+function Invoke-IndexerDetailNameUse
+{
+    param(
+        [Parameter(Mandatory)]
+        [PSCustomObject]
+        $indexer
+    )
+
+    return $(IF ($indexer.IndexerUrls) { '[' + ($indexer.name -replace $markdown_escape_regex, $markdown_escape_regex_rep) + '](' + ($indexer.IndexerUrls[0]) + ')' + '{#' + $indexer.infoLink.Replace($wiki_infolink.ToString(), '') + '}' } Else { IF ($indexer.fields.value[0]) { '[' + ($indexer.name -replace $markdown_escape_regex, $markdown_escape_regex_rep) + '](' + ($indexer.fields.value[0].Replace('api.', '').Replace('feed.', '')) + ')' + '{#' + $indexer.infoLink.Replace($wiki_infolink.ToString(), '') + '}' } Else { ($indexer.name -replace $markdown_escape_regex, $markdown_escape_regex_rep) + '{#' + $indexer.infoLink.Replace($wiki_infolink.ToString(), '') + '}' } } )
+}
+function Invoke-IndexerDetailNameTor
+{
+    param(
+        [Parameter(Mandatory)]
+        [PSCustomObject]
+        $indexer
+    )
+    {
+        return $(IF ($indexer.IndexerUrls) { '[' + ($indexer.name -replace $markdown_escape_regex, $markdown_escape_regex_rep) + '](' + ($indexer.IndexerUrls[0]) + ')' + '{#' + $indexer.infoLink.Replace($wiki_infolink.ToString(), '') + '}' } Else { ($indexer.name -replace $markdown_escape_regex, $markdown_escape_regex_rep) + '{#' + $indexer.infoLink.Replace($wiki_infolink.ToString(), '') + '}' }  )
```

Comment:
You should pass in those escape regex's or create them in the relevant functions that use them (scoping not an issue currently but scoping things correctly will help with any future changes and also for understanding of what is happening)